src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -924,12 +923,15 @@ export class ShelleyWallet implements WalletInterface {
     return txInfos.reduce((result, txInfo) => ({...result, [txInfo.id]: txInfo}), {} as Record<string, TransactionInfo>)
   }
 
-  async fetchUTXOs() {
+  async syncUtxos() {
     const addresses = [...this.internalAddresses, ...this.externalAddresses]
-    await this.utxoService.syncUtxoState(addresses)
-    const utxos = await this.utxoService.getAvailableUtxos()
 
-    return utxos.map(toUTXO)
+    await this.utxoManager.sync(addresses)
+
+    this._utxos = await this.utxoManager.getCachedUtxos()
+
+    // notifying always -> sync from lib need to flag if something has changed
+    this.notify({type: 'utxos', utxos: this.utxos})
```

Comment:
right now it is notifying always, I didn't want to subscribe to the storage, because the fetch and save are done in batches of 50 addresses, which will cause many events triggered like the current transactions, will update the lib later to return whether something updated or not, this will save all unnecessary renders.